lib/zone.ts

```diff
@@ -1318,13 +1318,24 @@ const Zone: ZoneType = (function(global: any) {
   function ZoneAwareError() {
     // Create an Error.
     let error: Error = NativeError.apply(this, arguments);
-
+    if (this && this !== global) {
```

Comment:
I think a better strategy would be to always assume that we were called without the `new` operator, that way we don't have to decide.